Amplify/Categories/DataStore/Query/QueryPredicate.swift

```diff
@@ -145,10 +146,15 @@ public class QueryPredicateOperation: QueryPredicate {
         if let intValue = value as? Int {
             return self.operator.evaluate(target: intValue)
         }
+
         if let timeValue = value as? Temporal.Time {
             return self.operator.evaluate(target: timeValue)
         }
 
+        if let enumValue = value as? EnumPersistable {
+            return self.operator.evaluate(target: enumValue.rawValue)
```

Comment:
in terms of other predicates like `greaterThan` or `lessThan`, the comparison will check for lexicographical ordering of `rawValue` of `Enum` right? It is not a valid usecase, but just wanted to understand the how it'd work

Reply:
yes that's correct, for example, the developer could take the string version of the enum and create a predicate like this for that check
```
let predicate = Post.keys.status.gt(PostStatus.draft.rawValue)
```

All posts with `.published` or `private` would match and is greater than `.draft`

Reply:
Added some more unit tests in here as "EnumToString" use cases

Reply:
LGTM